eli5/keras/explain_prediction.py

```diff
@@ -253,90 +451,253 @@ def _extract_image(doc):
     return image
 
 
-def _validate_doc(model, doc):
-    # type: (Model, np.ndarray) -> None
-    """
-    Check that the input ``doc`` is suitable for ``model``.
-    """
-    if not isinstance(doc, np.ndarray):
-        raise TypeError('doc must be a numpy.ndarray, got: {}'.format(doc))
-    input_sh = model.input_shape
-    doc_sh = doc.shape
-    if len(input_sh) == 4:
-        # rank 4 with (batch, ...) shape
-        # check that we have only one image (batch size 1)
-        single_batch = (1,) + input_sh[1:]
-        if doc_sh != single_batch:
-            raise ValueError('Batch size does not match (must be 1). ' 
-                             'doc must be of shape: {}, '
-                             'got: {}'.format(single_batch, doc_sh))
+def _unbatch_tokens(tokens):
```

Comment:
FWIW I think it's fine to require tokens to be a list without any extra dimentions, but current support also looks fine to me